server/metadata/database.go

```diff
@@ -23,3 +25,59 @@ type DatabaseMetadata struct {
 func (dm *DatabaseMetadata) SetDatabaseId(id uint32) {
 	dm.Id = id
 }
+
+const (
+	MainBranch          = "main"
+	BranchNamePrefix    = "$branch$_"
+	BranchNameSeparator = "_"
+)
+
+type DatabaseName struct {
+	db     string
+	branch string
+}
+
+func NewDatabaseNameWithBranch(db string, branch string) *DatabaseName {
+	if branch == MainBranch {
+		branch = ""
+	}
+	return &DatabaseName{
+		db:     db,
+		branch: branch,
+	}
+}
+
+func NewDatabaseName(key string) *DatabaseName {
+	if !strings.HasPrefix(key, BranchNamePrefix) {
+		return NewDatabaseNameWithBranch(key, "")
+	}
+	s := strings.SplitN(key, BranchNameSeparator, 3)
```

Comment:
If the database name starts with "_" then this will break? What if the naming convention is `$_{dbname}$branch_{branchName}` then it will be easy to extract first database name and then the branch name.